### PR TITLE
maint: update AWS credendtials workflow step from v1 -> v4

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
## Which problem is this PR solving?

Updates the AWS configure step used in the validate template workflow to use a newer version as v1 is now deprecated.

## Short description of the changes

- Update `configure-aws-credentials` step to use v4

## How to verify that this has the expected result

The workflow can now complete as it's using a supported version.

### CODE REVIEW

**Review by JamieDanielson — COMMENTED** [review]
Do we also need to update this in the [`publish-templates.yaml` workflow](https://github.com/honeycombio/cloudformation-integrations/blob/d5bf442b73c4e3b0b5d36e6ea84af204f36c9d14/.github/workflows/publish-templates.yml#L28)?

**Comment by dougireton** [comment]
Maybe you want to update `actions/checkout` to v4 as well?

https://github.com/search?q=repo:honeycombio/cloudformation-integrations%20actions/checkout&type=code

**Comment by JamieDanielson** [comment]
Hm, so it looks like the failed check from the other PR is actually on the publish-templates workflow based on the [job action run](https://github.com/honeycombio/cloudformation-integrations/actions/runs/9424415697/workflow) - not the validate workflow. And I don't think it's because of the deprecated action. The failure is in Configure AWS Credentials, which includes a secret. The workflow sets `pull_request`, not `pull_request_target`. The main difference between the two is that the [secrets are not passed to the runner when a workflow is triggered from a forked repository](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflows-in-forked-repositories). So I think the failure would be resolved if someone with write access on this repo committed a change to trigger that workflow.

Either way it's probably worth updating these actions - but I don't think that is the primary issue causing the checks to fail on the other PR.

**Review by JamieDanielson — APPROVED** [review]
Let's merge this in and see if it helps with the CI on the other PR.

**Pull request merged**
